packages/kbn-search-index-documents/components/result/rich_result_header.tsx

```diff
@@ -118,22 +120,39 @@ const MetadataPopover: React.FC<MetaDataProps> = ({
       </EuiFlexGroup>
       {onDocumentDelete && (
         <EuiPopoverFooter>
-          <EuiButton
-            iconType="trash"
-            color="danger"
-            size="s"
-            data-test-subj="deleteDocumentButton"
-            onClick={(e: React.MouseEvent<HTMLElement>) => {
-              e.stopPropagation();
-              onDocumentDelete();
-              closePopover();
-            }}
-            fullWidth
+          <EuiToolTip
+            content={
+              /* for serverless search users hasDeleteDocumentsPrivilege flag indicates if user has privilege to delete documents, for stack hasDeleteDocumentsPrivilege would be undefined */
+              hasDeleteDocumentsPrivilege !== undefined && !hasDeleteDocumentsPrivilege
```

Comment:
nitpick: `hasDeleteDocumentsPrivilege === false` would be more obvious condition

Reply:
updated in aaccd0bfe963c5be0817e64cbc28edfc2fad4193